nixos/modules/services/admin/nomachine.nix

```diff
@@ -0,0 +1,233 @@
+
+    EnablePasswordDB 0
+
+    Section "Server"
+
+    Name "Connection to localhost"
+    Host 127.0.0.1
+    Protocol NX
+    Port 4000
+    Authentication password
+
+    EndSection
+  '';
+
+  nodeConf = ''
+    ConfigFileVersion 4.0
+
+    EnableDebug ${toString cfg.enableDebug}
+    SessionLogLevel ${toString cfg.debugLevel}
+
+    AvailableSessionTypes unix-remote,unix-console,unix-default,unix-application,physical-desktop,shadow
+
+    AudioInterface pulseaudio
+
+    EnableSmartcardSharing 1
+    EnableCUPSSupport 0
+    CUPSBinPath ""
+    CUPSSbinPath ""
+    CUPSBackendPath ""
+
+    EnableEGLCapture 0
+
+    DisplayServerThreads auto
+    DisplayEncoderThreads auto
+    EnableDirectXSupport 0
+  '';
+in {
+  options.services.nxserver = {
+    enable = mkEnableOption "the NoMachine remote desktop server";
+
+    enableDebug = mkEnableOption "debug output";
+    debugLevel = mkOption {
+      default = 6;
+      example = 6;
+      description = "Debug output level";
+      type = lib.types.int;
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to open port 4000 in the firewall.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users = {
+      "nx" = {
+        group = "nx";
+        isSystemUser = true;
+        shell = "${nxserver}/NX/etc/nxserver";
+        home = "/var/lib/nxserver/nx";
+      };
+    };
+
+    users.groups = {
+      "nx" = {};
+    };
+
+    security.pam.services.nx.text = ''
+      # This is a default PAM configuration for NoMachine. It is based on
+      # system's 'su' configuration and can be adjusted freely according
+      # to administrative needs on the system.
+
+      auth       include       su
+      account    include       su
+      password   include       su
+      session    include       su
+    '';
+
+    security.pam.services.nxlimits.text = ''
+      # This is a default PAM configuration for NoMachine.
+      # Used to obtain nx user and nxhtd user limits.
+
+      session   optional pam_limits.so
+    '';
+
+    environment.etc = {
+      "NX/server/localhost/server.cfg" = {
+        text = ''ServerRoot = "${nxserver}/NX"'';
+        mode = "644";
+      };
+
+      "NX/server/localhost/node.cfg" = {
+        text = ''NodeRoot = "${nxserver}/NX"'';
+        mode = "644";
+      };
+
+      "NX/server/localhost/client.cfg" = {
+        text = ''ClientRoot = "${nxserver}/NX"'';
+        mode = "644";
+      };
+
+      "NX/nxserver".source = "${nxserver}/bin/nxserver";
+      "NX/nxnode".source = "${nxserver}/bin/nxnode";
+
+      "NX/server.cfg".text = serverConf;
+      "NX/node.cfg".text = nodeConf;
+    };
+
+    security.wrappers = {
+      nxexec = {
+        source = "${nxserver}/NX/bin/nxexec.orig";
+        owner = "root";
+        group = "root";
+        setuid = true;
+      };
+    };
+
+    networking.firewall.allowedTCPPorts = mkIf (cfg.openFirewall) [4000];
```

Comment:
Nit: put one space at each side of the port number, like [ 4000 ]

Reply:
I use [alejandra](https://github.com/kamadorueda/alejandra) for formatting which removes the spaces. Is there a specific formatting style that needs to be followed?

Reply:
There's no standard yet: https://github.com/NixOS/rfcs/pull/101